### ISSUE TITLE
Problem saving belongsToMany when using a Behaior that adds an order clause in beforeFind

### ISSUE THREAD

**Issue by veganista** [cakephp/cakephp#5962]
There seems to be a problem with adding an order clause in a beforeFind method from a Behvaior when saving a belongsToMany association.

Here is my set up.

__Behavior__
```php
/* OrderBehavior.php */
public function beforeFind(Event $event, Query $query, ArrayObject $options)
{
    if (!$query->clause('order')) {
        $query->order([$this->_table->alias() . '.order' => 'ASC']);
    }
}
```
__Associations__
```php
/* CoursesTable.php */
public function initialize(array $config)
{
    //...
    $this->belongsToMany('Modules', [
        'through' => 'course_modules',
    ]);
    //...
}

/* ModulesTable.php */
public function initialize(array $config)
{
    //...
    $this->belongsToMany('Courses', [
        'through' => 'course_modules',
    ]);
}

/* CourseModulesTable.php */
public function initialize(array $config)
{
    //..
    $this->belongsTo('Courses', [
        'foreignKey' => 'course_id'
    ]);
    $this->belongsTo('Modules', [
        'foreignKey' => 'module_id'
    ]);

    $this->addBehavior('Order');
}
```
When trying to save with this set up you get the following SQL which is incorrect.
```sql
SELECT `course_modules`.`course_id` AS `course_modules__course_id`, `course_modules`.`module_id` AS `course_modules__module_id`, `course_modules`.`order` AS `course_modules__order` FROM `course_modules` `course_modules` WHERE (`course_id` = :c0 AND `module_id` = :c1) ORDER BY `course_modules`.`order` ASC
UNION SELECT `course_modules`.`course_id` AS `course_modules__course_id`, `course_modules`.`module_id` AS `course_modules__module_id`, `course_modules`.`order` AS `course_modules__order` FROM `course_modules` `course_modules` WHERE (`course_id` = :c2 AND `module_id` = :c3) ORDER BY `course_modules`.`order` ASC
```

This produces the following error:
```
 Error: SQLSTATE[HY000]: General error: 1221 Incorrect usage of UNION and ORDER BY 
```
The above SQL can be successfully ran if it looked like (note the parentheses around each `SELECT`):
```sql
(SELECT `course_modules`.`course_id` AS `course_modules__course_id`, `course_modules`.`module_id` AS `course_modules__module_id`, `course_modules`.`order` AS `course_modules__order` FROM `course_modules` `course_modules` WHERE (`course_id` = :c0 AND `module_id` = :c1) ORDER BY `course_modules`.`order` ASC)
UNION (SELECT `course_modules`.`course_id` AS `course_modules__course_id`, `course_modules`.`module_id` AS `course_modules__module_id`, `course_modules`.`order` AS `course_modules__order` FROM `course_modules` `course_modules` WHERE (`course_id` = :c2 AND `module_id` = :c3) ORDER BY `course_modules`.`order` ASC)
```
This is explained here: http://dev.mysql.com/doc/refman/5.0/en/union.html

**Comment by markstory** [comment]
It looks like we've always stripped out the () out of union queries but I don't think there will be problems re-introducing them as () are supported in all RDBMS.

**Comment by markstory** [comment]
Well I was wrong about all RDBMS supporting () in Unions. SQLite does not support them unfortunately.

**Comment by markstory** [comment]
This should be fixed with #5967
